Source/WebCore/rendering/RenderLayerBacking.cpp

```diff
@@ -1705,6 +1717,10 @@ void RenderLayerBacking::updateAfterDescendants()
         m_graphicsLayer->setContentsOpaque(!m_hasSubpixelRounding && m_owningLayer.backgroundIsKnownToBeOpaqueInRect(compositedBounds()));
     }
 
+    if (layerRendererStyleHas3DTransformOperation(m_owningLayer)
+        || m_owningLayer.hasCompositedScrollableOverflow())
+        m_graphicsLayer->markDamageRectsUnreliable();
```

Comment:
So, what we want is to make sure that the whole layer is damaged, no? What about using setShouldDamageFullLayer or something similar instead of marking damage rects as unreliable?

Reply:
The whole *frame* needs to be invalidated, not the layer being checked.